# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/media/ipc/RemoteDecoderManagerParent.cpp
# Commit: c9f9bba02acf
# Full Hash: c9f9bba02acf40a824a92fd9430cc5e10edcc485
# Author: Matt Woodrow <mwoodrow@mozilla.com>
# Date: 2019-06-28 16:12:48
# Regressor Bug: 1561178
# File Overlap Count: 1
# Description:
#   Bug 1561178 - Make passing a TextureFactoryIdentifier optional, and pass one whenever we can. r=jya
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D35965
# ==============================================================================

diff -r c633a515314d -r c9f9bba02acf dom/media/ipc/RemoteDecoderManagerParent.cpp
--- a/dom/media/ipc/RemoteDecoderManagerParent.cpp	Fri Jun 28 06:31:59 2019 +0000
+++ b/dom/media/ipc/RemoteDecoderManagerParent.cpp	Fri Jun 28 06:32:16 2019 +0000
@@ -190,7 +190,7 @@
 PRemoteDecoderParent* RemoteDecoderManagerParent::AllocPRemoteDecoderParent(
     const RemoteDecoderInfoIPDL& aRemoteDecoderInfo,
     const CreateDecoderParams::OptionSet& aOptions,
-    const layers::TextureFactoryIdentifier& aIdentifier, bool* aSuccess,
+    const Maybe<layers::TextureFactoryIdentifier>& aIdentifier, bool* aSuccess,
     nsCString* aBlacklistedD3D11Driver, nsCString* aBlacklistedD3D9Driver,
     nsCString* aErrorDescription) {
   RefPtr<TaskQueue> decodeTaskQueue =